### PR TITLE
Support for exam import

### PR DESCRIPTION
**Pull request by flexik**
I've implemented support for importing exams as events from KOSapi into Sirius. Import of students associated with exams is also included.

### CODE REVIEW

File: app/interactors/fetch_exam_students.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      exam
+    end
+  end
+
+  def results
+    { events: @exams, people: @people.values }
+  end
+
+  def load_exam_events(faculty_semester)
+    Event.where(event_type: 'exam', faculty: faculty_semester.faculty, semester: faculty_semester.code)#.where('starts_at > NOW()')
```
Comment (jirutka) [inline app/interactors/fetch_exam_students.rb]: This comment `#.where('starts_at > NOW()')` looks like forgotten from testing…

File: app/interactors/fetch_exams.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require 'interactors/kosapi_interactor'
+
+class FetchExams < KOSapiInteractor
+
+  def perform(faculty_semester:, limit: 100, paginate: true)
+    @rooms = {}
+    @faculty_semester = faculty_semester
+    client = kosapi_client(faculty_semester)
+    @exams = client.exams.where("course.faculty==#{faculty_semester.faculty};semester=#{faculty_semester.code}").limit(limit)
```
Comment (jirutka) [inline app/interactors/fetch_exams.rb]: Please always prefer double equal sign (`==`) before a single one in RSQL.

File: app/interactors/convert_exams.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      events: @events,
+      courses: @courses.values,
+      people: @people.values
+    }
+  end
+
+  def convert_exam(exam)
+    event = Event.new
+    event.starts_at = exam.start_date
+    event.ends_at = (exam.end_date ? exam.end_date : exam.start_date + DEFAULT_EXAM_DURATION)
```
Comment (jnv) [inline app/interactors/convert_exams.rb]: I'd extract this logic to a separate method.

**Comment by jnv** [comment]
It's huge but seems fine, can we push this to staging (if it isn't there already) so we see how it performs?

I just dislike the inline logic for default event length and type, it seems to me this should be at least a separate method, or better yet, handled in a delegator for the special exam import role.

File: app/interactors/convert_exams.rb (hunk 3 above)
Comment (flexik) [inline app/interactors/convert_exams.rb]: I've created special role `ImportedExam` for dealing with some of the import logic and refactored the method under question.

File: app/interactors/fetch_exams.rb (hunk 2 above)
Comment (flexik) [inline app/interactors/fetch_exams.rb]: Fixed in 40684db40f06d71ba9904a22042d85e2885b05d6.

File: app/interactors/fetch_exam_students.rb (hunk 1 above)
Comment (flexik) [inline app/interactors/fetch_exam_students.rb]: This was supposed to be a disabled optimization. I've parametrized `FetchExamStudents` to allow enabling this as an option.

File: app/interactors/kosapi_interactor.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+require 'sirius/kosapi_client_registry'
+
+class KOSapiInteractor
+  include Interpipe::Interactor
+
+  def setup(client: nil)
+    @forced_client = client
+  end
+
+  protected
```
Comment (jirutka) [inline app/interactors/kosapi_interactor.rb]: Please add space after `protected`.

**Comment by jirutka** [comment]
:shipit:

**Pull request merged**